--- tuxonice-userui-0.7.2.orig/Makefile	2007-07-22 11:22:42.000000000 +0200
+++ tuxonice-userui-0.7.2/Makefile	2007-07-22 11:24:14.000000000 +0200
@@ -2,8 +2,8 @@
 LDFLAGS :=
 
 DESTDIR :=
-PREFIX := /usr/local
-INSTDIR := $(DESTDIR)$(PREFIX)/sbin
+PREFIX := /usr
+INSTDIR := $(DESTDIR)$(PREFIX)/lib/tuxonice-userui
 
 MODULES = text fbsplash usplash
 TARGETS = $(patsubst %,tuxoniceui_%,$(MODULES))
