Description: Do not link against libz and liblcms.
Author: Andrey Rahmatullin <wrar@wrar.name>
Last-Update: 2012-03-05

diff --git a/Makefile b/Makefile
index 553a9ff..dd9e088 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ LIBS =
 # FBSPLASH
 ifdef USE_FBSPLASH
 OBJECTS += fbsplash
-LIBS += -lmng -lpng -ljpeg -lz -lfreetype -llcms -lm
+LIBS += -lmng -lpng -ljpeg -lfreetype -lm
 LIB_TARGETS = fbsplash/userui_fbsplash.o
 CFLAGS += -DUSE_FBSPLASH
 endif
